### PR TITLE
Fix pyarrow and dask parquet issues

### PR DESCRIPTION
**Pull request by ianthomas23**
This is the final set of fixes identified in issue #86. CI works locally for me now.

The changes are all related to API changes and deprecations in pyarrow between 5.0.0 and 8.0.0. Changes were required in our code to directly deal with these as well as other changes following dask modifications to deal with the same.

I have created a new script `_create_testdata.py` to create test parquet files that are stored in the new `tests/test_data` directory and these are checked as part of `pytest`. The last time the CI definitely worked was July 2021 with `pyarrow==5.0.0` and `dask==2021.7.2` (and the same for `distributed`). These files are successfully read with up-to-date `pyarrow==8.0.0` and `dask==2022.7.1`. Similarly, test parquet files create with the up-to-date `pyarrow` and `dask` are successfully read with `pyarrow==5.0.0` and `dask==2021.7.2`.

### CODE REVIEW

**Comment by ianthomas23** [comment]
CI passing on github actions on ubuntu using Python 3.8. Expanding test matrix to cover Python 3.7 to 3.10 and all 3 major platforms.

**Review by jbednar — APPROVED** [review]
Thanks!

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -28,8 +28,8 @@ jobs:
         shell: bash -l {0}
     env:
       CHANS_DEV: "-c pyviz/label/dev  -c conda-forge"
-      PKG_TEST_PYTHON: "--test-python=py37"
-      PYTHON_VERSION: "3.7"
+      PKG_TEST_PYTHON: "--test-python=py39"
+      PYTHON_VERSION: "3.9"
```
Comment (maximlt) [inline .github/workflows/build.yaml:32]: There's a known issue on pyctdev that occurs when building a package with a base Python version greater than 3.7. You might hit that :/

**Review by maximlt — COMMENTED** [review]
Left a comment.

File: .github/workflows/build.yaml (hunk 1 above)
Comment (ianthomas23) [reply .github/workflows/build.yaml:32]: Thanks, I will revert to 3.7 for the time being then.

**Pull request merged**